File: java/src/test/java/problems1/ValidSudokuTest.java
package problems1;

import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ValidSudokuTest extends SampledTest {
    @Override
    public String getFileName() {
        return "valid_sudoku.json";
    }

    @Test
    public void test() {
        for (ValidSudoku solution : ValidSudoku.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isValidSudoku(sample.input.board))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Boolean> {
        public static class Input {
            public char[][] board;
        }
    }
}

File: typescript/__tests__/problems1_100/valid-sudoku.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/valid-sudoku';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    board: string[][],
  },
  output: boolean,
  message: string,
}

describe(
    'ValidSudoku',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        strictEqual(
                            solution.isValidSudoku(sample.input.board),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
